streams/src/test/java/org/apache/kafka/streams/tests/SmokeTestDriver.java

```diff
@@ -252,20 +189,32 @@ private static void shuffle(final int[] data, final int windowSize) {
         }
     }
 
-    public static void verify(final String kafka, final Map<String, Set<Integer>> allData, final int maxRecordsPerKey) {
+    public static boolean verify(final String kafka, final Map<String, Set<Integer>> allData, final int maxRecordsPerKey) {
         final Properties props = new Properties();
         props.put(ConsumerConfig.CLIENT_ID_CONFIG, "verifier");
         props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, kafka);
-        props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class);
+        props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
         props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class);
 
-        final KafkaConsumer<byte[], byte[]> consumer = new KafkaConsumer<>(props);
-        final List<TopicPartition> partitions = getAllPartitions(consumer, "echo", "max", "min", "dif", "sum", "cnt", "avg", "wcnt", "tagg");
+        final KafkaConsumer<String, byte[]> consumer = new KafkaConsumer<>(props);
+        final List<TopicPartition> partitions = getAllPartitions(consumer, "data", "echo", "max", "min", "dif", "sum", "cnt", "avg", "wcnt", "tagg");
         consumer.assign(partitions);
         consumer.seekToBeginning(partitions);
 
         final int recordsGenerated = allData.size() * maxRecordsPerKey;
         int recordsProcessed = 0;
+        final Map<String, AtomicInteger> processed = Stream.of("data", "echo", "max", "min", "dif", "sum", "cnt", "avg", "wcnt", "tagg").collect(Collectors.toMap(t -> t, t -> new AtomicInteger(0)));
+
+        final HashMap<String, LinkedList<Integer>> dataEvents = new HashMap<>();
+        final HashMap<String, LinkedList<Integer>> echoEvents = new HashMap<>();
+        final HashMap<String, LinkedList<Integer>> maxEvents = new HashMap<>();
+        final HashMap<String, LinkedList<Integer>> minEvents = new HashMap<>();
+        final HashMap<String, LinkedList<Integer>> difEvents = new HashMap<>();
+        final HashMap<String, LinkedList<Long>> sumEvents = new HashMap<>();
+        final HashMap<String, LinkedList<Long>> cntEvents = new HashMap<>();
+        final HashMap<String, LinkedList<Double>> avgEvents = new HashMap<>();
+        final HashMap<String, LinkedList<Long>> wcntEvents = new HashMap<>();
+        final HashMap<String, LinkedList<Long>> taggEvents = new HashMap<>();
```

Comment:
Why do we need all output events? Also, if we store all output events, why do we still need the current once from below? Seems redundant?

Reply:
Also for debugging. If a key has the wrong value, we can compare the input and output events to see where it went wrong.

Yes, it's redundant with the last result below, I can take it out.